Installer.php

```diff
@@ -79,33 +79,41 @@ public function getDownloadUrl( PackageEvent $event ) {
 		$package           = $this->getPackageFromOperation( $event->getOperation() );
 		$plugin            = false;
 		$package_name      = $package->getName();
+		$package_version   = $package->getPrettyVersion();
 
 		switch ( $package_name ) {
 
 			case 'junaidbhura/advanced-custom-fields-pro':
-				$plugin = new Plugins\AcfPro( $package->getPrettyVersion() );
+				$plugin = new Plugins\AcfPro( $package_version );
 				break;
 
 			case 'junaidbhura/polylang-pro':
-				$plugin = new Plugins\PolylangPro( $package->getPrettyVersion() );
+				$plugin = new Plugins\PolylangPro( $package_version );
 				break;
 
 			case 'junaidbhura/wp-all-import-pro':
 			case 'junaidbhura/wp-all-export-pro':
-				$plugin = new Plugins\WpAiPro( $package->getPrettyVersion(), str_replace( 'junaidbhura/', '', $package_name ) );
+				$plugin = new Plugins\WpAiPro( $package_version, str_replace( 'junaidbhura/', '', $package_name ) );
 				break;
 
 			default:
 				if ( 0 === strpos( $package_name, 'junaidbhura/gravityforms' ) ) {
-					$plugin = new Plugins\GravityForms( $package->getPrettyVersion(), str_replace( 'junaidbhura/', '', $package_name ) );
+					$plugin = new Plugins\GravityForms( $package_version, str_replace( 'junaidbhura/', '', $package_name ) );
 				} elseif ( 0 === strpos( $package_name, 'junaidbhura/wpai-' ) ) {
-					$plugin = new Plugins\WpAiPro( $package->getPrettyVersion(), str_replace( 'junaidbhura/', '', $package_name ) );
+					$plugin = new Plugins\WpAiPro( $package_version, str_replace( 'junaidbhura/', '', $package_name ) );
 				}
 
 		}
 
 		if ( ! empty( $plugin ) ) {
 			$this->downloadUrl = $plugin->getDownloadUrl();
+
+			$package_url = $package->getDistUrl();
+			if ( false !== strpos( $package_url, '%version%' ) ) {
+				$package->setDistUrl( str_replace( '%version%', $package_version, $package_url ) );
+			} elseif ( false === strpos( $package_url, $package_version ) ) {
+                $package->setDistUrl( $package_url . '#v' . $package_version );
```

Comment:
Given how generic the initial `dist.url` between WP plugin packages and how certain classes from `WPProPlugins\Plugins\*` support add-ons, we end up with dist URLs and cache keys such as:

- `junaidbhura/gravityforms`:
  - Dist URL: `https://www.gravityforms.com/#v2.4.20.3`
  - Cache Path: `junaidbhura/gravityforms/335917ed5650d888fcf4ad07a379d42d76e43b7b.zip`
- `junaidbhura/gravityformsmailchimp`:
  - Dist URL: `https://www.gravityforms.com/#v4.7.1`
  - Cache Path: `junaidbhura/gravityformsmailchimp/67b72a2a27e2aed83e0af5d338b5799fa73045c2.zip`

If you eventually support a private WP plugin where add-ons share the same version as the main package (like Symfony does between all its components), then we will end up with the same cache key across packages. They ZIPs are stored in different directories, nonetheless this could be problematic.

It may be a better choice to use `PackageInterface::getUniqueName()` (includes vendor, name, and version) instead of `PackageInterface::getPrettyVersion()` (just the version).

Reply:
Alternatively, we could delegate the filtering of the `dist.url` to the plugin classes themselves:

```php
class AcfPro {
    // ...
	/**
	 * Get the dist URL for this plugin.
	 *
	 * @return string
	 */
	public function getDistUrl() {
		return 'https://www.advancedcustomfields.com/#v' . $this->version;
	}
    // ...
}
```

This would require some more significant changes to this library but may be easier to maintain long-term.

Reply:
Hey @mcaskill - first of all - thanks for finding the problem! 🎉 

Do you think we could solve the problem by using a hash instead of the version number? So something like `https://www.gravityforms.com/#v<base64_encode( $package_name . $package_version )>` . That way even if the addons have the same versions, there would be no conflict?

Reply:
Yeah, we could hash the unique name. Should be okay.

Reply:
Do you want the hashing to occur in the current (4a6744e) proposal or in the [delegated dist URL](https://github.com/junaidbhura/composer-wp-pro-plugins/pull/20#discussion_r484113895) concept?

Reply:
Hey @mcaskill lets do it with the current proposal for now! Thanks!